### PR TITLE
Migrate `spglib` to new `SpglibDataset` format with version 2.5.0

### LINKED ISSUE #3922: Adapt pymatgen.symmetry.analyzer to new spglib (>=2.5.0) output data format
### Feature Requested

The recent spglib release v2.5.0 changed the output data types from dict to new classes like SpglibDataset.
https://spglib.readthedocs.io/en/stable/releases.html#v2-5-0-9-jul-2024

The dict-like interface is planned to be removed, so it will be necessary to adapt the pymatgen interface in the near future.

### Proposed Solution

For now, modifiying the version of spglib to <2.5.0 in pyproject.toml might be best - @DanielYang59 already confirmed that this fixes the failing `test_get_crystal_system` in tests/symmetry/test_analyzer.py (https://github.com/materialsproject/pymatgen/pull/3859#issuecomment-2228279612), so that test can be re-included @shyuep 

### Relevant Information

_No response_

### PR DESCRIPTION
**Pull request by DanielYang59**
### Summary

- Migrate `spglib` to new data format as pointed out by @kaueltzen, to fix #3922
- Add types for `SpacegroupAnalyzer` and `PointGroupAnalyzer`

### CODE REVIEW

File: tests/symmetry/test_analyzer.py [hunk 1]
```diff
@@ -170,12 +170,11 @@ def test_get_crystal_system(self):
         assert crystal_system == "orthorhombic"
         assert self.disordered_sg.get_crystal_system() == "tetragonal"
 
-        # orig_spg = self.sg._space_group_data["number"]
-        # self.sg._space_group_data["number"] = 0
-        # with pytest.raises(ValueError, match="Received invalid space group 0"):
-        #     self.sg.get_crystal_system()
-        #
-        # self.sg._space_group_data["number"] = orig_spg
+    @pytest.mark.skip("TODO: does not support item assignment")
```
Comment (DanielYang59) [inline tests/symmetry/test_analyzer.py:173]: The new `SpglibDataset` is frozen:

```
@dataclasses.dataclass(eq=False, frozen=True)
class SpglibDataset(DictInterface):
    """Spglib dataset information.

    .. versionadded:: 1.9.4
        The member 'choice' is added.
    .. versionadded:: 2.5.0
    """
```

**Comment by DanielYang59** [comment]
@kaueltzen Is everything looking good to you?

**Review by janosh — APPROVED** [review]
thanks a lot @DanielYang59! 👍 those deprecation warnings were spamming my logs

**Comment by DanielYang59** [comment]
Thanks for reviewing. @janosh For some reason [all tests in windows are failing](https://github.com/materialsproject/pymatgen/actions/runs/9957894146/job/27510952746)? Doesn't looks like owing to this PR and still need to find out why I guess

**Pull request merged**